components/create_post/create_post.jsx

```diff
@@ -348,22 +353,26 @@ export default class CreatePost extends React.Component {
     }
 
     sendMessage = (post) => {
-        post.channel_id = this.props.currentChannel.id;
+        const {
+            actions,
+            currentChannel,
+            currentUserId,
+            draft
+        } = this.props;
+
+        post.channel_id = currentChannel.id;
 
         const time = Utils.getTimestamp();
-        const userId = this.props.currentUserId;
+        const userId = currentUserId;
         post.pending_post_id = `${userId}:${time}`;
         post.user_id = userId;
         post.create_at = time;
         post.parent_id = this.state.parentId;
 
-        GlobalActions.emitUserPostedEvent(post);
+        actions.onSubmitPost(post, draft.fileInfos);
 
-        createPost(post, this.props.draft.fileInfos, () => {
-            GlobalActions.postListScrollChange(true);
-            this.setState({
-                submitting: false
-            });
+        this.setState({
```

Comment:
I think this state change must wait until the onSubmitPost finishes, currently I think it doesn't wait.

Reply:
Yeah, it's true. I could add a callback function for it like:
```
actions.onSubmitPost(post, draft.fileInfos, () => {
    this.setState({submitting; false});
});
```
but we are setting it to false anyway. So I did that way. Let me know which do you prefer.